src/backend/access/transam/xact.c

```diff
@@ -2317,14 +2317,14 @@ CommitTransaction(void)
 	willHaveObjectsFromSmgr =
 			PersistentEndXactRec_WillHaveObjectsFromSmgr(EndXactRecKind_Commit);
 
-	if (willHaveObjectsFromSmgr)
-	{
-		/*
-		 * We need to ensure the recording of the [distributed-]commit record and the
-		 * persistent post-commit work will be done either before or after a checkpoint.
-		 */
-		CHECKPOINT_START_LOCK;
-	}
+	/* In previous version, we ensured the recording of the [distributed-]commit record and the
+	 * persistent post-commit work will be done either before or after a checkpoint.
+	 *
+	 * However the persistent table status will be synchronized with AOSeg_XXXX
+	 * table and hdfs file in PersistentRecovery_Scan() at recovery PASS2.
+	 * We don't need to worry about inconsistent states between them. So no
+	 * CHECKPOINT_START_LOCK any more.
+	 */
```

Comment:
Not true. Consider this case.

1) flush commit record.
2) start a checkpoint.
3) checkpoint complete successfully.
4) failed to drop file and or fail to modify persistent table.

Since checkpoint was started (2) and finished (3) successfully after flush commit record (1), the commit record will be truncated during the checkpoint but failure (4) actually happened. I do not think recovery process can handle this case well.